### PR TITLE
properly support watchdog 4+ where events always have dest_path

### PR DESCRIPTION
**Pull request by evgeni**
### Pull Request Checklist

- [x] I’ve read the [guidelines for contributing](https://github.com/getnikola/nikola/blob/master/CONTRIBUTING.rst).
- [x] I updated AUTHORS.txt and CHANGES.txt (if the change is non-trivial) and documentation (if applicable).
- [x] I tested my changes.

### Description

Since watchdog 4 all events have the `dest_path` property,
but that property can be *empty*, thus breaking `nikola auto`:

    ERROR: asyncio: Task exception was never retrieved
    future: <Task finished name='Task-57' coro=<CommandAuto.reload_page() done, defined at nikola/plugins/command/auto/__init__.py:427> exception=ValueError('no path specified')>
    Traceback (most recent call last):
      File "nikola/plugins/command/auto/__init__.py", line 440, in reload_page
        p = os.path.relpath(event_path, os.path.abspath(self.site.config['OUTPUT_FOLDER'])).replace(os.sep, '/')
      File "/usr/lib64/python3.8/posixpath.py", line 453, in relpath
        raise ValueError("no path specified")
    ValueError: no path specified

### CODE REVIEW

**Comment by evgeni** [comment]
See https://github.com/gorakhargosh/watchdog/commit/41fca1eb601139922817a17b6335a8c2d2c8db51 for the change in watchdog

**Comment by evgeni** [comment]
Fixes https://github.com/getnikola/nikola/issues/3766, I guess

File: nikola/plugins/command/auto/__init__.py [hunk 1]
```diff
@@ -429,7 +429,7 @@ async def reload_page(self, event) -> None:
         # Move events have a dest_path, some editors like gedit use a
         # move on larger save operations for write protection
         if event:
-            event_path = event.dest_path if hasattr(event, 'dest_path') else event.src_path
+            event_path = event.dest_path if (hasattr(event, 'dest_path') and event.dest_path) else event.src_path
```
Comment (evgeni) [inline nikola/plugins/command/auto/__init__.py:432]:
```suggestion
            event_path = getattr(event, 'dest_path', None) or event.src_path
```
This was suggested in #3766 and also works, not sure which is more Pythonic tho.

Comment (Kwpolska) [reply nikola/plugins/command/auto/__init__.py:432]: I think the suggested change is less Pythonic.

**Review by Kwpolska — APPROVED** [review]

**Comment by Kwpolska** [comment]
Thanks for contributing!

**Pull request merged**